### PR TITLE
Fix conversion

### PR DESCRIPTION
**Pull request by bandi13**
Fix issues related to `-Wconversion` and `-Wsign-conversion`. Most notably where a variable needs to be cast to a different type. Not including implicit type upgrades that happen in equations (ie: uint8 + uint16).

### CODE REVIEW

**Comment by dgarske** [comment]
@bandi13 , seems like: `./configure CFLAGS="-Wconversion -Wsign-conversion"` does a good job reproducing the warnings. This PR does not resolve them all, should I expect a certain section to be resolved? I believe you are planning to split up the work into multiple PR's. Do we really care about resolving these warnings outside the library proper (like API unit test and examples)? Seems like `size_t` is used enough places that we can assume it's available, right? Looks like it comes from `stddef.h`.

**Review by dgarske — APPROVED** [review]
Lots of great cleanups for implicit casts. I don't foresee any portability issues. Will wait to merge until your response on my comment.

**Comment by bandi13** [comment]
Oh for sure there are a ton more to do. I was basically going through by type of error, not based on file the error occurs in. At some point I ended up scripting some of the changes.

Ultimately, I'd like to have `-Wconversion` tests enabled on PRs so we don't get into this mess again. It doesn't take very long to run, and it could run as a GitHub Action.

File: src/ssl.c [hunk 1]
```diff
@@ -1690,7 +1690,7 @@ const char* wolfSSL_get_shared_ciphers(WOLFSSL* ssl, char* buf, int len)
         return NULL;
 
     cipher = wolfSSL_get_cipher_name_iana(ssl);
-    len = min(len, (int)(XSTRLEN(cipher) + 1));
+    len = (int)min((word32)len, (int)(XSTRLEN(cipher) + 1));
```
Comment (dgarske) [inline src/ssl.c:1693]: Our MIN() uses word32, however I'm not sure that is true for all cases. But for our testing purposes this should be fine.

**Review by dgarske — APPROVED** [review]

**Pull request merged**